Library/Homebrew/cmd/log.rb

```diff
@@ -5,9 +8,11 @@ def log
       exec "git", "log", *ARGV.options_only
     else
       begin
-        path = ARGV.formulae.first.path
+        path = Formulary.canonical_path(ARGV.named.first)
+      # Maybe the formula was deleted
+      rescue TapFormulaUnavailableError => e
+        path = HOMEBREW_LIBRARY/"Taps/#{e.user}/homebrew-#{e.repo}/#{e.shortname}.rb"
```

Comment:
Can we add the path to the exception object where it is raised, instead of reconstructing it here?